src/sentry/tsdb/base.py

```diff
@@ -197,49 +197,42 @@ def normalize_ts_to_rollup(self, epoch, seconds):
         """
         return int(epoch / seconds)
 
-    def get_optimal_rollup(self, start_timestamp, end_timestamp):
-        """
-        Identify the lowest granularity rollup available within the given time
-        range.
-        """
-        num_seconds = int(to_timestamp(end_timestamp)) - int(to_timestamp(start_timestamp))
-
-        # This loop attempts to find the smallest possible rollup that will
-        # contain both the start and end timestamps. ``self.rollups`` is
-        # ordered from the highest resolution (smallest interval) to lowest
-        # resolution (largest interval.)
-        # XXX: There is a bug here, since this function assumes that the end
-        # timestamp is always equal to or greater than the current time. If the
-        # time range is shifted far enough into the past (e.g. a 30 second
-        # window, retrieved several days after it's occurrence), this can
-        # return a rollup that has already been evicted due to TTL, even if a
-        # lower resolution representation of the range exists.
-        for rollup, samples in six.iteritems(self.rollups):
-            if rollup * samples >= num_seconds:
-                return rollup
-
-        # If nothing actually matches the requested range, just return the
-        # lowest resolution interval.
-        return list(self.rollups)[-1]
+    def get_optimal_rollup(self, start):
+        """
+        Return the size (in seconds) of the finest-granularity available rollup
+        that will have data available in the given time range.
```

Comment:
This is a much better explanation, good clarification. Though, it's not a time range any longer, I guess, so it might make sense to say "available between the provided start time and the current timestamp"?